src/core/maps/cell_occupancy_estimator.h

```diff
@@ -47,6 +56,80 @@ struct Occupancy {
 struct Beam {
   double x_st, y_st; ///< Coordinates of a start of the beam.
   double x_end, y_end; ///< Coordinates where the beam reaches an obstacle.
+
+  //methods
+  bool tangents_rect(const Rectangle& bnds) const {
+    return equal(x_st, x_end, bnds.left) || equal(y_st, y_end, bnds.bot) ||
+           equal(x_st, x_end, bnds.right) || equal(y_st, y_end, bnds.top);
+  }
+
+  bool intersects_rect(const Rectangle& bnds) const {
+    bool result = false;
+    if(bnds.contains(x_st, y_st) || bnds.contains(x_end, y_end)) {
+      return true;
+    }
+    if (equal(x_st, x_end)) {
+      return(is_between(bnds.bot, y_st, y_end) &&
+             is_strictly_between(x_st, bnds.left, bnds.right));
+    }
+
+    if(both_less(x_st, x_end, bnds.left) ||
+       both_higher(x_st, x_end, bnds.right) ||
+       both_less(y_st, y_end, bnds.bot) ||
+       both_higher(y_st, y_end, bnds.top)) {
+         return false;
+    }
+    double k = (y_end - y_st) / (x_end - x_st);
```

Comment:
The code, you've implemented relies on enum class Intersections. For the purpose of this function it is redundant